ikarus/utils/modalanalysis/modalanalysis.hh

```diff
@@ -0,0 +1,228 @@
+#include <ikarus/utils/concepts.hh>
+#include <ikarus/utils/modalanalysis/lumpingschemes.hh>
+#include <ikarus/utils/modalanalysis/modalanalysishelper.hh>
+#include <ikarus/utils/modalanalysis/tags.hh>
+
+namespace Ikarus::Dynamics {
+
+/**
+ * \brief Opinionated wrapper class for GeneralizedSymEigenSolver suited for modal analysis
+ *
+ * \tparam FEC the type of container for finite elements
+ * \tparam DV the type of the DirichletValues
+ */
+template <typename FEC, typename DV>
+struct ModalAnalysis
+{
+  using Assembler     = SparseFlatAssembler<FEC&, DV>;
+  using MatrixType    = typename Assembler::MatrixType;
+  using ScalarType    = typename MatrixType::Scalar;
+  using FERequirement = typename Assembler::FERequirement;
+  using FEContainer   = FEC;
+
+  using LumpedAssembler =
+      AssemblerManipulator<Assembler, Ikarus::Impl::AssemblerInterfaceHelper<ScalarAssembler, ScalarManipulator>,
+                           Ikarus::Impl::AssemblerInterfaceHelper<VectorAssembler, VectorManipulator>,
+                           Ikarus::Impl::AssemblerInterfaceHelper<MatrixAssembler, MatrixManipulator>>;
+  using Solver = GeneralizedSymEigenSolver<EigenValueSolverType::Spectra, MatrixType>;
+
+  /**
+   * \brief Construct a new Modal Analysis object
+   *
+   * \tparam FES deduced type of the container of the finite elements passed to the constructor
+   * \param fes the container of the finite elements
+   * \param dv the DirichletValues
+   */
+  template <typename FES>
+  ModalAnalysis(FES&& fes, const DV& dv)
+      : fes_(std::forward<FES>(fes)),
+        d_(dv.basis().size()),
+        dRef_(d_),
+        stiffAssembler_(std::make_shared<Assembler>(fes_, dv)),
+        massAssembler_(std::make_shared<Assembler>(fes_, dv)) {
+    if constexpr (std::remove_cvref_t<decltype(fes.front())>::Traits::useEigenRef)
+      req_.insertGlobalSolution(dRef_);
+    else
+      req_.insertGlobalSolution(d_);
+
+    stiffAssembler_->bind(req_, Ikarus::AffordanceCollections::elastoStatics, Ikarus::DBCOption::Reduced);
+    massAssembler_->bind(req_, Ikarus::AffordanceCollections::dynamics, Ikarus::DBCOption::Reduced);
+    lumpedMassAssembler_ = makeAssemblerManipulator(*massAssembler_);
+  }
+
+  /**
+   * \brief Binds a lumping scheme for the mass matrix. The Lumping scheme should have an operator () and modify the
+   * matrix mat. For details see already existing lumping schemes at \file ikarus/utils/modalanalysis/lumpingschemes.hh.
+   * It resets already bound matrix manipulation functions.
+   *
+   * \tparam LumpingScheme The type of the lumping scheme, for example one found at \file
+   * ikarus/utils/modalanalysis/lumpingschemes.hh.s
+   * \param ls the instantiated LumpingScheme object (pass either by value or by template definition).
+   */
+  template <typename LumpingScheme>
+  void bindLumpingScheme(LumpingScheme ls = {}) {
+    lumpedMassAssembler_->unbindAllMatrixFunctions();
```

Comment:
I think for this specific case its okay, as the assembler is freshly instantiated in the class and is not meant to be used outside of this class. 
But in general it would be nice to have a mechanism to unbind certain functions from the manipulator

Reply:
Ah yes, I missed the fact that `lumpedMassAssembler_` is a `private` variable. Then nevermind here. And, if the user calls `unbindAllMatrixFunctions`, then they must be aware that all bound functions are being reset here.